### PR TITLE
Add/delete reactions

### PR DESCRIPTION
**Pull request by rebornix**
In next Insider build (as there is no Insider build for today), you should be able to add/delete comment from the comment title action bar or inline

kapture 2019-02-14 at 15 06 13

### CODE REVIEW

File: src/github/pullRequestManager.ts [hunk 1]
```diff
@@ -844,6 +844,44 @@ export class PullRequestManager {
 		return parseGraphQLComment(comment);
 	}
 
+	async addCommentReaction(pullRequest: PullRequestModel, graphNodeId: string, reaction: vscode.CommentReaction): Promise<AddReactionResponse> {
+		let reactionEmojiToContent = getReactionGroup().reduce((prev, curr) => {
+			prev[curr.label] = curr.title;
+			return prev;
+		}, {} as { [key: string]: string });
+		const { mutate } = await pullRequest.githubRepository.ensure();
+		const { data } = await mutate<AddReactionResponse>({
```
Comment (RMacfarlane) [inline src/github/pullRequestManager.ts]: Do we intend to also support reactions for GHE?

File: src/view/reviewDocumentCommentProvider.ts [hunk 2]
```diff
@@ -718,6 +726,57 @@ export class ReviewDocumentCommentProvider implements vscode.DocumentCommentProv
 		});
 	}
 
+	public async addReaction(document: vscode.TextDocument, comment: vscode.Comment, reaction: vscode.CommentReaction) {
+		await this.editReaction(document, comment, reaction, true);
+	}
+
+	public async deleteReaction(document: vscode.TextDocument, comment: vscode.Comment, reaction: vscode.CommentReaction) {
```
Comment (RMacfarlane) [inline src/view/reviewDocumentCommentProvider.ts]: I keep getting `Deleting the comment reaction failed: Illegal argument: line must be non-negative.` when trying to remove a comment while a PR is checked out

File: src/github/pullRequestManager.ts (hunk 1 above)
Comment (rebornix) [reply src/github/pullRequestManager.ts]: We don't. This API won't be called when there is no graphql support.

File: src/view/reviewDocumentCommentProvider.ts (hunk 2 above)
Comment (rebornix) [reply src/view/reviewDocumentCommentProvider.ts]: I digged into this bug and turns out to be not related to this pr. It's triggerred by the comments update service, I'll work on another PR to fix this.

**Comment by rebornix** [comment]
@RMacfarlane thanks for the review! The error is triggerred by another bug we introduced before so I'd like to fix that separately.

**Pull request merged**